news/backends/sfmc.py

```diff
@@ -11,19 +12,54 @@
 
 
 class ETRefreshClient(ET_Client):
+    token_cache_prefix = 'backends:sfmc:auth:'
+    token_property_names = [
+        'authToken',
+        'authTokenExpiration',
+        'internalAuthToken',
+        'refreshKey',
+    ]
+    _old_authToken = None
+    _old_authTokenExpiration = None
+    _old_internalAuthToken = None
+    _old_refreshKey = None
+
+    def _cache_key(self, prop):
+        return self.token_cache_prefix + prop
+
+    def _refresh_auth_tokens(self):
+        """Refresh the auth token and other values from cache"""
+        if self.authToken is not None and time() + 300 < self.authTokenExpiration:
+            # no need to refresh if the current tokens are still good
+            return
+
+        for prop in self.token_property_names:
+            setattr(self, prop, cache.get(self._cache_key(prop)))
+            # set the value so we can detect if it changed later
+            setattr(self, '_old_' + prop, getattr(self, prop))
+
+    def _cache_auth_tokens(self):
+        for prop in self.token_property_names:
+            new_val = getattr(self, prop)
+            if new_val is not None and new_val != getattr(self, '_old_' + prop):
+                cache.set(self._cache_key(prop), new_val, 3600)
```

Comment:
What is the advantage of iterating through all of the token_property_names and storing them under individual cache keys vs storing all of them in a dict with a single cache key?